### ISSUE TITLE
Create Organization Security Bug

### ISSUE THREAD

**Issue by kb-0311** [PalisadoesFoundation/talawa-api#1061]
**Describe the bug**

In the talawa-admin portal the users needs to be a verified `SUPERADMIN` to get access to the portal logically. In the Dashboard of talawa-admin Portal , there is a create organization function.

In short a user needs to be a super admin to create an organization according to the logic. However, in the backend no check is provided for the user role before creating an organization , thus enabling any one (USER, ADMIN , SUPERADMIN etc )to create an 
 organization.

This is a security bug.

This is security by obfuscation where capabilities are not presented to the user, but are still available.

**To Reproduce**

1. Sign up as a user in graphql playground 
2. get the access token
3. Create and organization through its mutation
4. An org is created

**Expected behavior**
Only super admins should be allowed to create new organizations. 
**Actual behavior**
A clear and concise description of how the code performed w.r.t expectations.
Right now anyone can technically create new organizations.
**Screenshots**

**Additional details**

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359
//SIgn up as a brand new default user
![Screenshot from 2023-02-12 11-39-24](https://user-images.githubusercontent.com/96020697/218296386-c23edd54-db0a-4e6c-89dd-c57f211928ae.png)
//Logged in as that user
![Screenshot from 2023-02-12 11-39-33](https://user-images.githubusercontent.com/96020697/218296369-88a3b058-7566-474c-befe-174753fe3aa0.png)
//  Created a brand new organization as `userType:USER`
![Screenshot from 2023-02-12 11-39-41](https://user-images.githubusercontent.com/96020697/218296392-08e8aa7a-4ecc-48a6-955f-6e0e1d5d49c2.png)
![Screenshot from 2023-02-12 11-41-19](https://user-images.githubusercontent.com/96020697/218296395-fa34af66-0ffc-4cad-9e7e-163cf7328090.png)
![Screenshot from 2023-02-12 11-41-48](https://user-images.githubusercontent.com/96020697/218296399-40fdef75-1bf4-439a-a7b5-7b4cd518fb3d.png)

**Comment by kb-0311** [comment]
pls assign this to me

**Comment by anshgoyalevil** [comment]
@kb-0311 You may resolve this issue using the ```roleDirective```

Just apply the role directive on the createOrganization mutation like so

```
type Query {
    hello: String @role(requires: "ADMIN")
  }
```

**Comment by kb-0311** [comment]
@anshgoyalevil Yes but that will not be enough from a modular development perspective imo.
1. The roleDirective will eventually be implemented for every query/Mutation that requires it. But as you can see in all typeDefs the role directive is not implemented for any directive that requires it. So while I am fixing security flaws I need to make sure to implement them one by one to not cause breaking change. 
3. There needs to be a thorough approach to how I am fixing security-related bugs consistent with the codebase so that other developers can easily debug and write tests in the future. That is why it is important to add a layer of check in the resolver itself as it has been done for all resolvers till now.

 Along with roleDirectives in gql I will also be adding an extra utility `checkSuperAdmin.ts` within `/utilities` just like `checkAdmin.ts` which currently exists, to add a more modular approach and make this function logically reusable whenever needed.

Also speaking of the roleDirectives, there is a slight mistake with the thrown error->
 
![Screenshot from 2023-02-12 12-53-42](https://user-images.githubusercontent.com/96020697/218298545-68a796ad-57e6-4edb-8d94-d7a123e8a952.png)

the error should be `'user.notAuthorised'`. 'user.notAuthenticated' is an error thrown by auth directive(which could get very confusing with manual testing and debugging the app) so will be fixing that too in this same PR. and add this as a constant.

**Comment by xoldyckk** [comment]
@kb-0311 You're changing the errors thrown by the server. If the clients check the error for different types to show a different UI you'll be breaking their functionality. Make sure to remember this.

**Comment by kb-0311** [comment]
@xoldyckk Yes, I will make sure to change the talawa-admin app and talawa-mobile app accordingly.

**Comment by kb-0311** [comment]
@xoldyckk could you pls remove the unapproved label too ?
